include: Remove compat include headers

galak opened the pull request:
Remove the compat headers as its been 2 releases since we introduced
them.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Remove compat include headers

Remove the compat headers as its been 2 releases since we introduced
them.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <misc/FOO.h> -> <sys/FOO.h>

Fix #include <misc/FOO.h> as misc/FOO.h has been deprecated and
should be #include <sys/FOO.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif commented:
hmm, looks like the deprecation warnings were not working correctly?

galak committed:
include: Fix use of <gpio.h> -> <drivers/gpio.h>

Fix #include <gpio.h> as it has been deprecated and
should be #include <drivers/gpio.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <i2c.h> -> <drivers/i2c.h>

Fix #include <i2c.h> as it has been deprecated and
should be #include <drivers/i2c.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <sys_io.h> -> <sys/sys_io.h>

Fix #include <sys_io.h> as it has been deprecated and
should be #include <sys/sys_io.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <json.h> -> <data/json.h>

Fix #include <json.h> as it has been deprecated and
should be #include <data/json.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <atomic.h> -> <sys/atomic.h>

Fix #include <atomic.h> as it has been deprecated and
should be #include <sys/atomic.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <flash_map.h> -> <storage/flash_map.h>

Fix #include <flash_map.h> as it has been deprecated and
should be #include <storage/flash_map.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

pabigot reviewed (commented):
I guess this is technically ok, but it would be more "polite" to have a release where the default is disabled, so people get some warning about the change before they're forced to rework all their source.

galak committed:
include: Fix use of <fs.h> -> <fs/fs.h>

Fix #include <fs.h> as it has been deprecated and
should be #include <fs/fs.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <nvs/nvs.h> -> <fs/nvs.h>

Fix #include <nvs/nvs.h> as it has been deprecated and
should be #include <fs/nvs.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
west.yml: Update modules for include fixes

Update west.yml to point to newer versions of fatfs, nffs, and mcumgr
to pull in fixes for using proper include files.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ulfalizer commented on west.yml:
```diff
@@ -38,7 +38,7 @@ manifest:
       path: modules/hal/esp-idf
     - name: fatfs
-      revision: df96914dd989907f3a5de4cb95b116c2f305820d
+      revision: pull/2/head
```
Are these leftovers?

ulfalizer replied:
Nm... guess it should be DNM until those are in at least.

galak committed:
include: Fix use of <power.h> -> <power/power.h>

Fix #include <power.h> as it has been deprecated and
should be #include <power/power.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <base64.h> -> <sys/base64.h>

Fix #include <base64.h> as it has been deprecated and
should be #include <sys/base64.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
include: Fix use of <crc.h> -> <sys/crc.h>

Fix #include <crc.h> as it has been deprecated and
should be #include <sys/crc.h>.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
